### ISSUE TITLE
How to automatically enable extensions

### ISSUE THREAD

**Issue by pitwegner** [CrunchyData/postgres-operator#2652]
I have recently deployed a Postgres cluster including timescale extension using PGO. However, I had to manually `CREATE EXTENSION IF NOT EXISTS timescaledb`. Is there a recommended way how to automate such a task? I initially thought of an initContainer, but this seems a little bulky and I expect such a routine requirement to be handled by the operator.

My current config:
```
apiVersion: postgres-operator.crunchydata.com/v1beta1
kind: PostgresCluster
metadata:
  name: test
spec:
  image: registry.developers.crunchydata.com/crunchydata/crunchy-postgres-ha:centos8-13.4-4.7.2
  postgresVersion: 13
  instances:
    - name: instance1
      replicas: 3
      resources:
        limits:
          cpu: "1"
          memory: 1Gi
        requests:
          cpu: 250m
          memory: 512Mi
      dataVolumeClaimSpec:
        accessModes:
        - "ReadWriteOnce"
        resources:
          requests:
            storage: 8Gi
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - weight: 1
            podAffinityTerm:
              topologyKey: kubernetes.io/hostname
              labelSelector:
                matchLabels:
                  postgres-operator.crunchydata.com/cluster: test
                  postgres-operator.crunchydata.com/instance-set: instance1
  backups:
    pgbackrest:
      image: registry.developers.crunchydata.com/crunchydata/crunchy-pgbackrest:centos8-2.33-1
      repoHost:
        dedicated: {}
      repos:
      - name: repo1
        volume:
          volumeClaimSpec:
            accessModes:
            - "ReadWriteOnce"
            resources:
              requests:
                storage: 20Gi
  monitoring:
    pgmonitor:
      exporter:
       image: registry.developers.crunchydata.com/crunchydata/crunchy-postgres-exporter:ubi8-5.0.2-0
  patroni:
    dynamicConfiguration:
      postgresql:
        parameters:
          shared_preload_libraries: timescaledb
```

**Comment by jkatz** [comment]
We've discussed allowing for extensions to be enabled automatically during bootstrap from the custom resource and certainly something to consider. Similarly, there is a feature in an upcoming release that will allow you to provide a SQL script at bootstrap/initialization that will let you run any commands you want at initialization (e.g. `CREATE EXTENSION IF NOT EXISTS timescaledb;`).

Presently, there are a few ways that you can enable an extension, each with its own automation path:

1. Create a [Postgres superuser](https://access.crunchydata.com/documentation/postgres-operator/v5/tutorial/user-management/#creating-a-new-user). Log into the database you want to enable the extension in. Create said extension. If need be, [delete the superuser](https://access.crunchydata.com/documentation/postgres-operator/v5/tutorial/user-management/#deleting-a-user)

2. Create a [Postgres superuser](https://access.crunchydata.com/documentation/postgres-operator/v5/tutorial/user-management/#creating-a-new-user). Enable said extension as a "[trusted extension](https://www.postgresql.org/docs/current/sql-createextension.html)".  If need be, [delete the superuser](https://access.crunchydata.com/documentation/postgres-operator/v5/tutorial/user-management/#deleting-a-user). Allow your users to create the extension as needed.

3. Use `kubectl exec` to create the the extension in a DB, e.g.

```
kubectl exec -n $NAMESPACE -it \
    $(kubectl -n postgres-operator get pods --selector=postgres-operator.crunchydata.com/role=master,postgres-operator.crunchydata.com/cluster=$CLUSTER_NAME -o name) \
    -- psql $DATABASE_NAME -c 'CREATE EXTENSION ...'
```

tl;dr, there will be at least one way to more easily automatically adding an extension to the DB in a future release, but there are a few ways you can do it now.

**Comment by cortopy** [comment]
The new feature looks very nice. A good idea would be also to have a `databaseUpdateSQL` so that it runs after an update. This would allow for example:

```
ALTER EXTENSION timescaledb UPDATE;
```
